packages/jest-mock/__typetests__/mock-functions.test.ts

```diff
@@ -492,3 +494,20 @@ expectError(
     (key: string, value: number) => {},
   ),
 );
+
+// mockProperty
+
+const obj = {
+  fn: () => {},
+
+  property: 1,
+};
+
+expectType<MockedProperty<number>>(mockProperty(obj, 'property', 1));
+mockProperty(obj, 'property', 1).mockValue(1).mockRestore();
```

Comment:
Might be TypeScript will not see difference between a property and a getter. But can a user see if `process.argv` is a property or a getter? Would it be possible to support getters? Or there are some limitations?

Reply:
Right, but simply looking at it how can I know which API to use: `spyOn` or `mockValue`?

Reply:
From my little perspective, there is unfortunately a reliance on internal implementation of a property when trying to mock it:

* `spyOn()` allows to mock _methods_ and provides rich interface to mock/intercept its calls
* `mockProperty()`'s idea is to mock dummy properties, and that's why it supports just dummy interface that's way leaner from what `spyOn()` returns.

When `mockProperty()` should support getters/setters, it raises these questions:
* Should `mockProperty()` be just proxy to `spyOn()` for getters/setters? 
* How should the call look? Should we allow mocking a value instead of a getter/setter? Wouldn't it break existing object shape as an accident and potentially break something inside?
* If getters/setters will be allowed to be mocked by both `mockProperty/spyOn`, what would be the _source of truth_, the single correct approach advised in the documentation? 

I do not mind extending the PR, but as there is right now already existing, well-known and rich way to mock getters/setters, I wanted to keep `mockProperty()` not conflicting - as it could be always enabled in the future, but if it is enabled from beginning, removing that functionality in future might be more difficult? However I might be wrong. Am really curious in your thoughts/directions here.

Reply:
Sure. It can be added later if there is a need.